### PR TITLE
EventEmitter - Error Handling

### PR DESCRIPTION
**Pull request by jlstendebach**
## Description
Added error handling to `EventEmitter.emit(type, event)`. It now aggregates the errors and throws them at the end of the emit call. This ensures two things:
1. All event listeners are invoked during the event emission process, regardless of errors thrown.
2. Listener lists are properly cleaned up.

Also added tests to handle edge cases and errors.

### CODE REVIEW

File: tests/events/EventEmitter.test.js [hunk 1]
```diff
@@ -406,5 +406,35 @@ describe("EventEmitter", () => {
...
+            const emitter = new EventEmitter();
+            const error = new Error("listener failure");
+            const callback = () => {
+                throw error;
+            };
+
+            emitter.once("tick", callback);
+
+            expect(() => emitter.emit("tick", {})).toThrow(error);
+            expect(emitter.getListenerCount("tick")).toBe(0);
```
[the remark this answers is not in the document]
Comment (jlstendebach) [reply tests/events/EventEmitter.test.js:444]: Added change in a separate commit.
